packages/venia-ui/lib/util/apolloCache.js

```diff
@@ -54,6 +55,9 @@ export const cacheKeyFromType = object => {
         // Only maintain a single cart entry
         case MagentoGraphQLTypes.Cart:
             return 'Cart';
+        // Use email address instead of deprecated id
+        case MagentoGraphQLTypes.Customer:
+            return `${MagentoGraphQLTypes.Customer}:${object.email}`;
```

Comment:
Here's a thought - perhaps we don't actually need an ID for caching since we only should store data for an authorized customer and clear it on log out, sort of like a cart.

So maybe like `Cart` above, we just return `Customer` for the Customer type. If we keep `email` as part of the id resolution it becomes this requirement that we can't really enforce without just knowing.

Reply:
Brilliant, didn't consider we should only ever have one Customer in the cache (until we get to B2B). I'll make this change 👍